beater/eventsapibeat.go

```diff
@@ -245,6 +272,58 @@ func (e *EventsAPIBeat) itemUsagesLoop(c chan<- *beat.Event) error {
 	}
 }
 
+func (e *EventsAPIBeat) auditEventsLoop(c chan<- *beat.Event) error {
+	ticker := time.NewTicker(e.config.AuditEvents.SampleFrequency)
+	defer ticker.Stop()
+
+	cursor, err := e.auditEventsCursorStore.GetValue()
+	if err != nil {
+		return fmt.Errorf("failed to get audit events cursor. %v", err)
+	}
+	if cursor == "" {
+		cursor = e.config.AuditEvents.StartingCursor
+	}
+
+	for {
+		select {
+		case <-ticker.C:
+			var errs []string
+
+			for {
+				auditEventsResponse, err := e.apiClient.AuditEvents(e.ctx, e.config.AuditEvents.AuthToken, cursor)
+				if err != nil {
+					errs = append(errs, fmt.Sprintf("failed to fetch audit events. %v", err))
+					break
+				}
+
+				for i := range auditEventsResponse.AuditEvents {
+					item := &auditEventsResponse.AuditEvents[i]
+
+					event := item.BeatEvent()
+					_, _ = event.PutValue("@metadata.event_type", AuditEventsType)
+
+					c <- event
+				}
+
+				cursor = fmt.Sprintf(`{ "cursor": "%s" }`, auditEventsResponse.Cursor)
+
+				if err := e.auditEventsCursorStore.SetValue(cursor); err != nil {
+					errs = append(errs, fmt.Sprintf("failed to set audit events cursor. %v", err))
+				}
+
+				if !auditEventsResponse.HasMore {
+					break
+				}
+
+			}
+
+			if len(errs) > 0 {
+				return fmt.Errorf(strings.Join(errs, "."))
+			}
+		}
+	}
+}
+
 func (e *EventsAPIBeat) Stop() {
 	e.cancel()
 	if e.signInAttemptsCursorStore != nil {
```

Comment:
should we add audit events to `Stop()` too?

Reply:
yeah definitely, I totally missed this, thanks for catching this!